src/Microsoft.DotNet.Build.CloudTestTasks/CreateAzureContainer.cs

```diff
@@ -3,110 +3,166 @@
-        /// When true, if the specified container already exists the task will fail.
-        /// </summary>
-        public bool FailIfExists { get; set; }
-
         public override bool Execute()
         {
-            Log.LogMessage(MessageImportance.High, "Creating container named '{0}' in storage account {1}.", ContainerName, AccountName);
-
-            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(string.Format("DefaultEndpointsProtocol=https;AccountName={0};AccountKey={1}", AccountName, AccountKey));
-            CloudBlobClient blobClient = storageAccount.CreateCloudBlobClient();
-            CloudBlobContainer newContainer = blobClient.GetContainerReference(ContainerName);
+            return ExecuteAsync().GetAwaiter().GetResult();
+        }
 
-            if (FailIfExists && newContainer.Exists())
-            {
-                Log.LogError("The container '{0}' already exists.", ContainerName);
-                return false;
-            }
-            else
+        public async Task<bool> ExecuteAsync()
+        {
+            Log.LogMessage(
+                MessageImportance.High, 
+                "Creating container named '{0}' in storage account {1}.", 
+                ContainerName, 
+                AccountName);
+            DateTime dt = DateTime.UtcNow;
+            string url = string.Format(
+                "https://{0}.blob.core.windows.net/{1}?restype=container", 
+                AccountName, 
+                ContainerName);
+            StorageUri = string.Format(
+                "https://{0}.blob.core.windows.net/{1}/", 
+                AccountName, 
+                ContainerName);
+
+            Log.LogMessage(MessageImportance.Normal, "Sending request to create Container");
+            using (HttpClient client = new HttpClient())
             {
-                newContainer.CreateIfNotExists();
+                using (HttpRequestMessage req = new HttpRequestMessage(HttpMethod.Put, url))
+                {
+                    req.Headers.Add(AzureHelper.DateHeaderString, dt.ToString("R", CultureInfo.InvariantCulture));
+                    req.Headers.Add(AzureHelper.VersionHeaderString, AzureHelper.StorageApiVersion);
+                    req.Headers.Add(AzureHelper.AuthorizationHeaderString, AzureHelper.AuthorizationHeader(
+                            AccountName,
+                            AccountKey,
+                            "PUT",
+                            dt,
+                            req));
+                    byte[] bytestoWrite = new byte[0];
+                    int bytesToWriteLength = 0;
+
+                    Stream postStream = new MemoryStream();
+                    postStream.Write(bytestoWrite, 0, bytesToWriteLength);
+                    req.Content = new StreamContent(postStream);
+                    
+                    HttpResponseMessage response = await client.SendAsync(req);
```

Comment:
This type is disposable too (for the same reasons HttpWebResponse is, and might cause the weird hangs again)